gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/resources/handler/authorization/AuthorizationRequestFailureHandler.java

```diff
@@ -227,7 +228,9 @@ private String buildRedirectUri(String error, String errorDescription, Authoriza
         query.computeIfAbsent(Parameters.STATE, val -> authorizationRequest.getState());
 
         boolean fragment = !isDefaultErrorPage(authorizationRequest.getRedirectUri(), errorPath) &&
-                (isImplicitFlow(authorizationRequest.getResponseType()) || isHybridFlow(authorizationRequest.getResponseType()));
+                (isImplicitFlow(authorizationRequest.getResponseType()) ||
+                        isHybridFlow(authorizationRequest.getResponseType()) ||
+                        ResponseMode.FRAGMENT.equals(authorizationRequest.getResponseMode()));
```

Comment:
suggestion: looks like the same condition is used in 3 places. maybe it can be encapsulated?